### DISCUSSION TITLE
Update a Segment via API call

### DISCUSSION THREAD

**Discussion by theomeuh** [Unleash/unleash#7980] [Q&A] [answered]
I'm trying to migrate some data we have in Unleash and I need to automatically update Segment via API. I'm referring to this endpoint: https://docs.getunleash.io/reference/api/unleash/update-segment

It says: "Any fields not specified will be left untouched", but if I try to PUT an empty body, I got a 400 Bad Request `{"id":"1dd37e9f-2b75-458b-b7ca-1424e0561479","name":"BadDataError","message":"Request validation failed: your request body or params contain invalid data. Refer to the `details` list for
more information.","details":[{"message":"\"name\" is required."}]}`

Then let's add a "dummy name".. I got the same message but this time, it is the field "constraints" which is apparently missing.

My question: is the doc up-to-date ? It really looks like a PUT (like the endpoint verb says) and not a PATCH (like the description says), do you confirm ?

PS: what I really want is to update "constraints" but unfortunately, I have the "name is required" error when I try to just push a payload with only "constraints"

**Comment by theomeuh** [comment]
I've played a little bit with the API, and it seems that since the Segment open api description says "name" is required, it is also required in this PUT endpoint, even though we want to update another param.

Should I create a bug issue to fix that ?

**Reply by FredrikOseberg** [reply]
@theomeuh This does indeed seem like a bug in the API. Please open an issue and we'll take care of it. Thank you!

**Reply by theomeuh** [reply, marked as answer]
I've created the following issue: https://github.com/Unleash/unleash/issues/7987

**Reply by theomeuh** [reply]
Can I let you put relevant labels on it ? I think I cannot :thinking:
